### DISCUSSION TITLE
Empty string in message.event field causes a runtime error.

### DISCUSSION THREAD

**Discussion by extremegf** [enisdenjo/graphql-sse#48] [Q&A] [answered]
[From RFC8895](https://www.rfc-editor.org/rfc/rfc8895.html#section-3.4)
>The protocol defines three field names: **event**, id, and data. If a message has more than one "data" line, the value of the data field is the concatenation of the values on those lines. There can be only one "event" and "id" line per message. The **"data" field is required**; the **others are optional**.

The check below however fails when `message.event === ""`

https://github.com/enisdenjo/graphql-sse/blob/b4019e45ad18e6b0b36bd9e5798909d9bb2599ec/src/parser.ts#L88

**Comment by enisdenjo** [comment, marked as answer]
The `event` field is required in the [GraphQL over SSE spec](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md) because it serves the purpose of describing the type of the message being sent.

**Reply by extremegf** [reply]
That would suggest that [Graphql Helix](https://github.com/contra/graphql-helix) is out of spec, by not sending `event: ` line. Shoud we open an issue at their side?

Relevant code: 
https://github.com/contra/graphql-helix/blob/1bbc7146e953556498180c53aa1638dd25072d43/packages/core/lib/send-result/node-http.ts#L91

**Comment by enisdenjo** [comment]
GraphQL Helix is out of spec because it doesn't aim to implement it at all.

Nevertheless, I recommend using the well-maintained [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server/) instead. It also has a [graphql-sse plugin](https://the-guild.dev/graphql/yoga-server/docs/features/subscriptions#graphql-over-server-sent-events-protocol-via-graphql-sse) that can get you going in no time.

**Reply by extremegf** [reply]
Yoga seems promising. Thanks!

**Reply by extremegf** [reply]
Hmm. Yoga doesn't set 'event:' eigher:

```
data: {"data":{"countdown":10}}

data: {"data":{"countdown":9}}

data: {"data":{"countdown":8}}
```

**Reply by extremegf** [reply]
Note that I did install the plugin. It doesn't seem to be necessary (or is in stalled by default) as it works the same without it.

**Reply by enisdenjo** [reply]
@extremegf are you connecting to the stream endpoint `/graphql/stream`?

**Reply by extremegf** [reply]
No. So there's a spearate stream endpoint? I'll look it up.

**Reply by enisdenjo** [reply]
Yes, its stated in the [graphql-sse plugin documentation](https://the-guild.dev/graphql/yoga-server/docs/features/subscriptions#graphql-over-server-sent-events-protocol-via-graphql-sse). You can of course configure the endpoint, but the default is `/graphql/stream`.

**Reply by extremegf** [reply]
Thank you for the help! I really appreciate it!
